synapse/util/caches/treecache.py

```diff
@@ -0,0 +1,52 @@
+SENTINEL = object()
+
+
+class TreeCache(object):
```

Comment:
Comments please.

- What does this do?
- What assumptions are there on the key/values?